packages/_shared/src/utils/entityHelpers.ts

```diff
@@ -305,3 +305,54 @@ export const getEntryImage = async (
     return null;
   }
 };
+
+const TRANSFORMATIONS_ENDPOINTS = {
+  'downloads.ctfassets.net': 'images.ctfassets.net',
+  'downloads.flinkly.com': 'images.flinkly.com',
+};
+
+const DOWNLOADS_ENDPOINTS = Object.keys(TRANSFORMATIONS_ENDPOINTS);
+
+type TransformationEndpointKey = keyof typeof TRANSFORMATIONS_ENDPOINTS;
+
+export const getResolvedImageUrl = (
+  url: string,
+  params?: { w?: number; h?: number; fit?: string },
+): string => {
+  try {
+    // Handle protocol-relative URLs by adding https: temporarily
+    const urlToParse = url.startsWith('//') ? `https:${url}` : url;
+    const parsedUrl = new URL(urlToParse);
+
+    // Replace downloads domain with images domain if it matches
+    if (DOWNLOADS_ENDPOINTS.includes(parsedUrl.hostname)) {
+      parsedUrl.hostname =
+        TRANSFORMATIONS_ENDPOINTS[parsedUrl.hostname as TransformationEndpointKey];
+    }
+
+    if (params) {
+      Object.entries(params).forEach(([key, value]) => {
+        if (value !== undefined) {
+          parsedUrl.searchParams.set(key, value.toString());
```

Comment:
I think `toString()` here is not needed as it's handled later already.
```suggestion
          parsedUrl.searchParams.set(key, value);
```